app/commands/cohort/join.rb

```diff
@@ -0,0 +1,15 @@
+class Cohort::Join
+  include Mandate
+
+  initialize_with :user, :cohort, :introduction
+
+  def call
+    # TODO: make this safe against race issues
+    user.cohort_memberships.create!(cohort:, status:, introduction:)
```

Comment:
I have a preference for:
```suggestion
    User::CohortMembership.create!(user: ,cohort:, status:, introduction:)
```